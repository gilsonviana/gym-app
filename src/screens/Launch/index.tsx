import * as React from 'react'
import { SafeAreaView, View, StyleSheet, TouchableOpacity, Image } from 'react-native'
import { StackNavigationProp } from '@react-navigation/stack'
import { FontAwesome5 } from '@expo/vector-icons';

import theme from '@styled/theme'
import Text from '@styled/Text'
import Button from '@styled/Button'

import { AuthStackParamList } from '@navigation/Auth'

interface Props {
    navigation: StackNavigationProp<AuthStackParamList, 'Launch'>
}

const LaunchScreen: React.FunctionComponent<Props> = ({
    navigation
}) => {
    const handleOnSubmit = (): void => {
        navigation.navigate('SignUp')
    }
    
    return (
        <View style={styles.container}>
            <Image testID="launch-background-image" blurRadius={.5} style={styles.img} source={{uri: 'https://images.unsplash.com/photo-1550345332-09e3ac987658?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1234&q=80'}} resizeMode="cover"/>
            <SafeAreaView style={{flex: 1}}>
                <View style={{flexDirection: 'row', justifyContent: 'center', marginTop: 10}}>
                    <FontAwesome5 name="dumbbell" size={24} color={theme.color.light} />
                    <Text light style={{marginLeft: 10}} large bold>GYM APP</Text>
                </View>
                <View style={{flex: 1, justifyContent: 'flex-end'}}>
                    <Text light large bold center style={{marginBottom: 10}}>Get trainer-guided workouts on demand</Text>
                    <Text light center style={{marginBottom: 20}}>Free 7-day trial</Text>
                    <Button text="Try it now" style={{marginBottom: 20}} onPress={handleOnSubmit}/>
                    <View style={{flexDirection: 'row', justifyContent: 'center', marginBottom: 40}}>
                        <Text light small>Already a member? </Text>
                        <TouchableOpacity hitSlop={{top: 5, bottom: 5, left: 5, right: 5}}>
                            <Text light small bold>log in</Text>
                        </TouchableOpacity>
                    </View>
                </View>
            </SafeAreaView>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: theme.color.light,
        paddingHorizontal: theme.spacing.margin,
    },
    img: {
        position: 'absolute',
        top: 0,
        bottom: 0,
        left: 0,
        right: 0
    }
})

export default LaunchScreen