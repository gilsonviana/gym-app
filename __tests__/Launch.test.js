import React from 'react'
import renderer from 'react-test-renderer'

import Launch from '@screens/Launch'

describe('<Launch />', () => {
    it('should render correctly', () => {
        const tree = renderer.create(<Launch />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})