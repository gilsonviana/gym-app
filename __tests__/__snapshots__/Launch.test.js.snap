// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Launch /> should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff8f0",
      "flex": 1,
      "paddingHorizontal": 20,
    }
  }
>
  <Image
    blurRadius={0.5}
    resizeMode="cover"
    source={
      Object {
        "uri": "https://images.unsplash.com/photo-1550345332-09e3ac987658?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1234&q=80",
      }
    }
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    testID="launch-background-image"
  />
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": 10,
        }
      }
    >
      <Text />
      <Text
        bold={true}
        large={true}
        light={true}
        style={
          Array [
            Object {
              "color": "#fff8f0",
              "fontSize": 18,
              "fontWeight": "bold",
              "textAlign": "left",
              "textDecorationColor": "black",
              "textDecorationLine": "none",
              "textDecorationStyle": "solid",
            },
            Object {
              "marginLeft": 10,
            },
          ]
        }
      >
        GYM APP
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "flex-end",
        }
      }
    >
      <Text
        bold={true}
        center={true}
        large={true}
        light={true}
        style={
          Array [
            Object {
              "color": "#fff8f0",
              "fontSize": 18,
              "fontWeight": "bold",
              "textAlign": "center",
              "textDecorationColor": "black",
              "textDecorationLine": "none",
              "textDecorationStyle": "solid",
            },
            Object {
              "marginBottom": 10,
            },
          ]
        }
      >
        Get trainer-guided workouts on demand
      </Text>
      <Text
        center={true}
        light={true}
        style={
          Array [
            Object {
              "color": "#fff8f0",
              "fontSize": 14,
              "fontWeight": "normal",
              "textAlign": "center",
              "textDecorationColor": "black",
              "textDecorationLine": "none",
              "textDecorationStyle": "solid",
            },
            Object {
              "marginBottom": 20,
            },
          ]
        }
      >
        Free 7-day trial
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#111d4a",
            "borderRadius": 375,
            "marginBottom": 20,
            "opacity": 1,
            "paddingBottom": 14,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 14,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff8f0",
                "textAlign": "center",
              },
            ]
          }
        >
          Try it now
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "center",
            "marginBottom": 40,
          }
        }
      >
        <Text
          light={true}
          small={true}
          style={
            Array [
              Object {
                "color": "#fff8f0",
                "fontSize": 12,
                "fontWeight": "normal",
                "textAlign": "left",
                "textDecorationColor": "black",
                "textDecorationLine": "none",
                "textDecorationStyle": "solid",
              },
            ]
          }
        >
          Already a member? 
        </Text>
        <View
          accessible={true}
          focusable={false}
          hitSlop={
            Object {
              "bottom": 5,
              "left": 5,
              "right": 5,
              "top": 5,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            bold={true}
            light={true}
            small={true}
            style={
              Array [
                Object {
                  "color": "#fff8f0",
                  "fontSize": 12,
                  "fontWeight": "bold",
                  "textAlign": "left",
                  "textDecorationColor": "black",
                  "textDecorationLine": "none",
                  "textDecorationStyle": "solid",
                },
              ]
            }
          >
            log in
          </Text>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
